crates/kubelet/src/state.rs

```diff
@@ -203,26 +216,42 @@ impl<PodState> Transition<PodState> {
     /// }
     /// ```
     #[allow(clippy::boxed_local)]
-    pub fn next<I: State<PodState>, S: State<PodState>>(_i: Box<I>, s: S) -> Transition<PodState>
+    pub fn next<I: State<ProviderState, PodState>, S: State<ProviderState, PodState>>(
+        _i: Box<I>,
+        s: S,
+    ) -> Transition<ProviderState, PodState>
     where
         I: TransitionTo<S>,
     {
         Transition::Next(StateHolder { state: Box::new(s) })
     }
+
+    #[allow(clippy::boxed_local)]
+    pub fn next_unchecked<S: State<ProviderState, PodState>>(
```

Comment:
Yes, but I can't get the exit from the last generic state to compile safely.  You end up with something like:

```
impl TransitionTo<Error> for ...;
impl TransitionTo<P::RunState> for ...;
```

And Rust complains you have conflicting implementations for `Error`, I think because it can't be sure that `P::RunState` and `Error` are different states.

If anyone knows a way round this I'm all ears!

Reply:
Yep, I ran into the same issue. It seems that P::RunState is interpreted as any state (which makes sense).